cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -174,6 +178,10 @@ public void start() {
     metricsCollectionService.startAndWait();
     serviceStore.startAndWait();
     leaderElection.startAndWait();
+
+    // Tries to create the ZK root node (which can be namespaced through the zk connection string)
+    Futures.getUnchecked(ZKOperations.ignoreError(zkClient.create("/", null, CreateMode.PERSISTENT),
```

Comment:
Probably we should check and add log the existence of the cdap root directory from the ZK connect (I think we could get it from config?) to make sure it is recorded we try to create root node.